src/shapes/src/Quarterpipe.cpp

```diff
@@ -0,0 +1,146 @@
+    auto y = pos_projected * Utils::vector_product(m_axis, m_orientation);
+    auto pos_delta_phi = std::atan2(y, x);
+
+    // point 1 and 2 are the end points of the shape where the cylinder
+    // intersects the cuboid. Their position vectors also define normals of the
+    // cuboid side planes 1 and 2
+    auto to_point_1 =
+        m_radius * vec_rotate(m_axis, -0.5 * m_delta_phi, m_orientation);
+    auto to_point_2 =
+        m_radius * vec_rotate(m_axis, 0.5 * m_delta_phi, m_orientation);
+
+    if (pos_delta_phi <= -0.5 * m_delta_phi) {
+      // Zone 1
+      vec = pos_projected - to_point_1;
+      dist = vec.norm();
+    } else if (pos_delta_phi >= 0.5 * m_delta_phi) {
+      // Zone 2
+      vec = pos_projected - to_point_2;
+      dist = vec.norm();
+    } else {
+      if (pos_norm < m_radius) {
+        // Zone 3
+        dist = m_radius - pos_norm;
+        vec = -dist * pos_projected / pos_norm;
+      } else {
+        // Zones 4 to 9 are hard to distingish, so all distances are calculated
+        // and the minimum chosen
+
+        // Zone 4, 5, 7 or 9
+        auto dist_plane_1 = (pos_projected * to_point_1 / to_point_1.norm()) -
+                            to_point_1.norm();
+        auto vec_plane_1 = dist_plane_1 * to_point_1 / to_point_1.norm();
+
+        // Zone 4, 6, 8 or 9
+        auto dist_plane_2 = (pos_projected * to_point_2 / to_point_2.norm()) -
+                            to_point_2.norm();
+        auto vec_plane_2 = dist_plane_2 * to_point_2 / to_point_2.norm();
+
+        // Distinguish the Zones
+        if (dist_plane_1 >= 0. and dist_plane_2 >= 0.) {
+          // Zone 9
+          auto corner_point = std::sqrt(2.) * m_radius * m_orientation;
+          vec = pos_projected - corner_point;
+          dist = vec.norm();
+        } else if (dist_plane_1 < 0. and dist_plane_2 < 0.) {
+          // Zone 4, 5 or 6
+
+          // First, find which side is closer
+          // Distinguishes zones 4 and 5 from 4 and 6
+          bool in_zone_4_or_5 =
+              Utils::abs(dist_plane_1) < Utils::abs(dist_plane_2);
+          auto dist_plane = in_zone_4_or_5 ? dist_plane_1 : dist_plane_2;
+          auto vec_plane = in_zone_4_or_5 ? vec_plane_1 : vec_plane_2;
+
+          // Now find if in 4 or 5/6
+          auto dist_cyl = m_radius - pos_norm;
+          auto vec_cyl = -dist_cyl * pos_projected / pos_norm;
+
+          bool in_zone_4 = Utils::abs(dist_cyl) < Utils::abs(dist_plane);
+          dist = in_zone_4 ? dist_cyl : dist_plane;
+          vec = in_zone_4 ? vec_cyl : vec_plane;
+        } else {
+          // Zone 7 or 8
+          bool in_zone_7 = dist_plane_1 >= 0;
+          dist = in_zone_7 ? dist_plane_1 : dist_plane_2;
+          vec = in_zone_7 ? vec_plane_1 : vec_plane_2;
+        }
+      }
+    }
+  } // 2D end, now to the 3rd dimension
+
+  auto upper_lower = rel_z > 0 ? 1 : -1;
+  // signed distance to the closer top/bottom wall
+  auto dist_cover = Utils::abs(rel_z) - 0.5 * m_height;
```

Comment:
you could replace the `Utils:abs` call which you have also done in all cases below
```suggestion
  auto dist_cover = upper_lower * rel_z - 0.5 * m_height;
```